test/functional/tests/volumes/test_many_lvms_on_single_core.py

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright(c) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from storage_devices.lvm import Lvm, LvmConfiguration
+from test_tools.lvm_common_utils import prepare_devices, get_test_configuration, run_fio_on_lvm
+from test_utils.size import Size, Unit
+
+lv_count = 16
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_many_lvms_on_single_core():
+    """
+        title: Test for LVM creation on CAS device - many lvms on single core.
+        description: |
+          Validation of LVM support, many LVMs (16) created on CAS device (1 cache, 1 core).
+        pass_criteria:
+          - CAS devices created successfully.
+          - LVMs created successfully.
+          - FIO with verification ran successfully.
+          - Configuration after reboot match configuration before.
+    """
+    with TestRun.step(f"Create CAS device."):
+
+        cache, core = prepare_devices(TestRun.disks['cache'], TestRun.disks['core'],
+                                      Size(8, Unit.GibiByte))
+
+    with TestRun.step("Create LVMs on CAS device."):
+        lvm_filters = ["a/.*/", "r|/dev/sd*|", "r|/dev/hd*|", "r|/dev/xvd*|", "r/disk/", "r/block/",
+                       "r|/dev/nvme*|"]
+        pv_num = 1
+        vg_num = 1
+        lv_num = 16
+        cache_num = 1
+        cas_dev_num = 1
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(core, config)
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(lvms)
+
+    with TestRun.step("Flush buffers"):
+        for lvm in lvms:
+            TestRun.executor.run(f"hdparm -f {lvm.path}")
+
+    with TestRun.step("Create init config from running configuration"):
+        config_before_reboot, devices_before = get_test_configuration()
+
+    with TestRun.step("Reboot system."):
+        TestRun.executor.reboot()
+
+    with TestRun.step("Validate running configuration"):
+        config_after_reboot, devices_after = get_test_configuration()
+
+        if config_after_reboot == config_before_reboot and devices_after == devices_before:
+            TestRun.LOGGER.info(f"Configuration is as expected")
+        else:
+            TestRun.LOGGER.error(f"Configuration changed after reboot.")
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(lvms)
+
+    with TestRun.step("Remove LVMs."):
```

Comment:
Not in this scenario, here LVMs are created on CAS device so first we have to remove LVMs (top "layer"), and just then stop the CAS (bottom "layer") which is not mentioned here since is included within cleanup